src/test/scala/chisel3/tests/ValidQueueTest.scala

```diff
@@ -0,0 +1,136 @@
+package chisel3.tests
+
+import chisel3._
+import chisel3.tester._
+import chisel3.experimental.BundleLiterals._
+import chisel3.tester.experimental.TestOptionBuilder._
+import chisel3.util.{Pipe, Valid}
+import org.scalatest._
+import treadle.VerboseAnnotation
+
+class ValidQueueModule(typeGen: Data, delay: Int) extends MultiIOModule {
+  val in = IO(Flipped(Valid(typeGen)))
+  val out = IO(Valid(typeGen))
+
+  out := Pipe(in, delay)
+}
+
+
+class ValidQueueTest extends FlatSpec with ChiselScalatestTester {
+  behavior of "Testers2 with ValidQueue"
+
+  it should "pass through elements, using enqueueNow" in {
+    test(new ValidQueueModule(UInt(8.W), delay = 3)) { c =>
+      c.in.initSource().setSourceClock(c.clock)
+      c.out.initSink().setSinkClock(c.clock)
+
+      c.out.expectInvalid()
+      c.in.enqueueNow(42.U)
+      parallel(
+          c.out.expectDequeue(42.U),
+          c.in.enqueueNow(43.U)
+      )
+      c.out.expectDequeue(43.U)
+    }
+  }
+
+  it should "pass through elements, using enqueueSeq" in {
+    test(new ValidQueueModule(UInt(8.W), delay = 3)).withAnnotations(Seq()) { c =>
+      c.in.initSource().setSourceClock(c.clock)
+      c.out.initSink().setSinkClock(c.clock)
+
+      c.out.expectInvalid()
+
+      fork {
+        c.in.enqueueSeq(Seq(42.U, 43.U, 44.U))
+      }.fork {
+        c.out.expectDequeue(42.U)
+        c.out.expectDequeue(43.U) // check that queue stalls
```

Comment:
Wait does the queue stall?
Also why not use dequeueSeq?

Reply:
queue did not stall - that was a leftover comment
now uses expectDequeueSeq